akka-actor/src/main/scala/akka/actor/FSM.scala

```diff
@@ -141,6 +145,7 @@ object FSM {
      */
     def forMax(timeout: Duration): State[S, D] = timeout match {
       case f: FiniteDuration ⇒ copy(timeout = Some(f))
+      case Duration.Inf      ⇒ copy(timeout = SomeMaxFiniteDuration)
       case _                 ⇒ copy(timeout = None)
```

Comment:
With the addition of `case Duration.Inf => ...`, isn't this line now dead code?

[Duration](http://www.scala-lang.org/api/2.11.5/index.html#scala.concurrent.duration.Duration)'s docs show that it has 2 children: `FiniteDuration` and `Infinite`.